docs/application/native/guides/device/diagnostics.md

```diff
@@ -0,0 +1,171 @@
+
+```cpp
+void crash_callback(diagnostics_ctx_h ctx, void *user_data)
+{
+    diagnostics_data_h data;
+    const char *params[] = {"cs_info_json"};
+
+    int ret = diagnostics_get_data(ctx, params, 1, &data);
+    if (ret != DIAGNOSTICS_ERROR_NONE)
+        return;
+
+    [...]
+}
+```
+
+Using the `params` argument it is possible to specify, which data must be read from the crash report. Currently, the proper values are:
+
+-   `"cs_info_json"`: JSON report is returned, which contains information about crashed process (callstack, mapped memory regions, and so on).
+-   `"cs_full"`: Full crash report ZIP archive is collected.
+
+The obtained data is associated with the `data` Diagnostics handler. To read the data content, you must use `diagnostics_data_read()`:
+
+```cpp
+void crash_callback(diagnostics_ctx_h ctx, void *user_data)
+{
+    diagnostics_data_h data;
+    const char *params[] = {"cs_info_json"};
+
+    int ret = diagnostics_get_data(ctx, params, 1, &data);
+    if (ret != DIAGNOSTICS_ERROR_NONE)
+        return;
+
+    char buff[0x1000];
+    int timeout_ms = 0;
+    size_t bytes_read;
+
+    for (;;) {
+        ret = diagnostics_data_read(data, buff, sizeof(buff), timeout_ms, &bytes_read);
+        if (ret == DIAGNOSTICS_ERROR_TRY_AGAIN)
+            continue;
+
+        if (ret != DIAGNOSTICS_ERROR_NONE)
+            break; // error occured
+
+        if (bytes_read == 0)
+            break; // no more data to read
+
+        fwrite(buff, 1, bytes_read, stdout); // write received data to the STDOUT
+    }
+}
+```
+
+## Diagnostic data request
+
+Diagnostics API can be used to request data from other application or service, especially for the diagnostic data that for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.
+
+To do so, you must call `diagnostics_request_client_data()`. The `client_id` variable must be the id of the process that supports Diagnostics API requests:
+
+```cpp
+void request_data()
+{
+    const char *client_id = "org.tizen.some_service";
+    diagnostics_data_h data;
+    const char *params[] = {"custom_parameter"};
+
+    int ret = diagnostics_request_client_data(client_id, params, 1, &data);
+    if (ret != DIAGNOSTICS_ERROR_NONE)
+        return;
+
+    char buff[0x1000];
+    int timeout_ms = 0;
+    size_t bytes_read;
+
+    for (;;) {
+        ret = diagnostics_data_read(data, buff, sizeof(buff), timeout_ms, &bytes_read);
+        if (ret == DIAGNOSTICS_ERROR_TRY_AGAIN)
+            continue;
+
+        if (ret != DIAGNOSTICS_ERROR_NONE)
+            break; // error occured
+
+        if (bytes_read == 0)
+            break; // no more data to read
+
+        fwrite(buff, 1, bytes_read, stdout); // write received data to the STDOUT
+    }
+}
+```
+
+If `diagnostics_request_client_data()` returns successfully, the obtained data can be read by `diagnostics_read_data()`.
```

Comment:
Just a question.

In the code, we have `diagnostics_data_read()`.
However in this description, `diagnostics_read_data()` is introduced.

Would you please check this mismatch?

Reply:
the description is wrong